lib/rules/no-array-constructor.js

```diff
@@ -107,6 +131,18 @@ module.exports = {
 				let fixText;
 				let messageId;
 
+				const hasSpreadArgument = node.arguments.some(
+					arg => arg.type === "SpreadElement",
+				);
+
+				const nonSpreadCount = node.arguments.filter(
+					arg => arg.type !== "SpreadElement",
+				).length;
+
+				const shouldSuggest =
+					(hasSpreadArgument && nonSpreadCount < 2) ||
+					hasCommentsInArrayConstructor(node);
```

Comment:
```suggestion
				const nonSpreadCount = node.arguments.reduce(
					(count, arg) =>
						arg.type !== "SpreadElement" ? count + 1 : count,
					0,
				);

				const shouldSuggest =
					(node.arguments.length > 0 && nonSpreadCount < 2) ||
					hasCommentsInArrayConstructor(node);
```

To avoid creating an intermediary array and looping over the array twice.